### ISSUE TITLE
Doesn't slide on Touch devices

### ISSUE THREAD

**Issue by aa09626** [vanderlee/limitslider#6]
Hello,

I've just discovered your slider extension and think it's great. I tried a demo page on my computer which worked as expected. But on a touch device (Galaxy Note 2), I cannot get the slide event to fire.

I am using 'jQuery UI Touch Punch' which works fine at detecting touch events on the regular jquery slider but for some reason it does not work with your extension. 

Would you happen to know why?

Thanks,
Alex

**Comment by vanderlee** [comment]
Limitslider is a very thin wrapper around jQuery-UI slider and doesn't handle any mouse activity itself, so it should work just fine. Perhaps jQueyr-UI touch punch has some specific handling for the slider, which doesn't extend to Limitslider. Could you please post a JSFiddle that demonstrates the issue?

**Comment by Johnafriedman** [comment]
Hi, thanks for the great slider extension. I am using it for a NFP literacy program for underserved students. This issue is the only one I have. Apparently the label rendering is preventing proper sliding. If you try to tap and drag on a mobile device, the handle will move only one step. Interestingly if you tap on the handle and then on a range of the same slider or a difference slider the slide occurs normally. You can see this behavior in this fiddle: http://jsfiddle.net/john_numina_org/b6dov9Lc/26/

Thanks Again,
John

**Comment by Johnafriedman** [comment]
Hi Again, this solution worked for me. 
```
		_renderLabel: function(index, init) {
			if (this.options.label) {
				var value = this.options.values[index];
				var text = $.isFunction(this.options.label) ? this.options.label(value, index) : value;

				var html = $('<div>').css({
					'text-align':		'center'
				,	'font-size':		'75%'
				,	'display':			'table-cell'
				,	'vertical-align':	'middle'
				}).html(text);

				if(init){
					$(this.handles[index]).html(html).css({
						'text-decoration':	'none'
						,	'display':			'table'
					});
				}else{
					$(this.handles[index]).find("div").text(text);
				}
			}
		},

		_renderLabels: function() {
			if (this.options.label) {
				var that = this;
				$.each(this.options.values, function(v) {
					that._renderLabel(v, true);
				});
			}
		},
```

**Comment by vanderlee** [comment]
Thanks for the code. I'll try to merge it in this weekend.

**Comment by vanderlee** [comment]
I made some minor changes to your code for performance, but other than that it is now merged in 1.3.1. Could you please verify?

**Comment by Johnafriedman** [comment]
Confirmed working. Thanks!

On Sat, Jun 25, 2016 at 10:19 AM, Martijn van der Lee <
notifications@github.com> wrote:

> I made some minor changes to your code for performance, but other than
> that it is now merged in 1.3.1. Could you please verify?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/vanderlee/limitslider/issues/6#issuecomment-228549965>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AGtnm0h7OzeKQKjmY0QRlSxzYELIKcHxks5qPUb8gaJpZM4DIu8l>
> .
>



-- 
*It All Starts With Literacy*

John Friedman
Software Developer
Innovations for Learning
www.innovationsforlearning.org
